File: app/src/main/java/com/robotnec/budget/activity/AddCategoryActivity.java
package com.robotnec.budget.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.robotnec.budget.R;
import com.robotnec.budget.core.domain.Category;
import com.robotnec.budget.fragments.AddCategoryFragment;
import com.robotnec.budget.util.Keys;

/**
 * @author zak <zak@swingpulse.com>
 */
public class AddCategoryActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (savedInstanceState == null) {
            Bundle args = getIntent().getExtras();
            Category category = (Category) args.getSerializable(Keys.CATEGORY);
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.frame_fragment, AddCategoryFragment.newInstance(category))
                    .commit();
        }
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_add_category;
    }
}

File: app/src/main/java/com/robotnec/budget/app/activity/AddCategoryActivity.kt
package com.robotnec.budget.app.activity

import android.os.Bundle
import com.robotnec.budget.R
import com.robotnec.budget.app.fragments.AddCategoryFragment
import com.robotnec.budget.app.util.Keys
import com.robotnec.budget.core.domain.Category

/**
 * @author zak zak@swingpulse.com>
 */
class AddCategoryActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (savedInstanceState == null) {
            val args = intent.extras
            val category = args.getSerializable(Keys.CATEGORY) as Category
            supportFragmentManager.beginTransaction()
                    .replace(R.id.frame_fragment, AddCategoryFragment.newInstance(category))
                    .commit()
        }
    }

    override fun layoutId(): Int {
        return R.layout.activity_add_category
    }
}
